tools/logparser2.py

```diff
@@ -29,68 +30,126 @@ def timestamps_gen(id, stream):
             if len(vals) != 2: # only at the end of file
                 return
             timestamp = float(vals[1])
+    stream.close()
 
 
-def sensor_gen(meta_filename, selected=None):
+class DirLog:
+    def __init__(self, dirpath):
+        self.dirpath = dirpath
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        pass
+
+    def meta(self):
+        filenames = os.listdir(self.dirpath)
+        meta_filename = [name for name in filenames if name.startswith('meta_')]
+        assert len(meta_filename) == 1
```

Comment:
Hmm, I didn't know that. So you cannot replay directory, you have to replay the meta file. Ok.